tests/KoPrivateGPT/retrieval/test_base_retrieval.py

```diff
@@ -47,13 +52,128 @@ def validate_passages(retrieved_passage: List[Passage], top_k: int):
+        }
+    },
+    {
+        'db_type': 'mongo_db',
+        'db_path': {
+            'mongo_url': 'test_url_2',
+            'db_name': 'test_db_name_2',
+            'collection_name': 'test_collection_name_2'
+        }
+    },
+    {
+        'db_type': 'mongo_db',
+        'db_path': {
+            'mongo_url': 'test_url_1',
+            'db_name': 'test_db_name_1',
+            'collection_name': 'test_collection_name_1'
+        }
+    }
+]
+
+TEST_DB_ORIGIN_RESULT = {(('db_type', 'mongo_db'), ('db_path', (('mongo_url', 'test_url_1'),
+                                                                ('db_name', 'test_db_name_1'),
+                                                                ('collection_name', 'test_collection_name_1')))): [0, 3],
+                         (('db_type', 'pickle_db'), ('db_path', (('save_path', 'test'),))): [1],
+                         (('db_type', 'mongo_db'), ('db_path', (('mongo_url', 'test_url_2'),
+                                                                 ('db_name', 'test_db_name_2'),
+                                                                 ('collection_name', 'test_collection_name_2')))): [2]}
+
+
+@pytest.fixture
+def just_bm25_retrieval():
+    bm25_path = os.path.join(root_dir, "resources", "bm25", "just_bm25_retrieval.pkl")
+    pickle_path = os.path.join(root_dir, "resources", "pickle", "just_bm25_retrieval.pkl")
+    if not os.path.exists(os.path.dirname(bm25_path)):
+        os.makedirs(os.path.dirname(bm25_path))
+    bm25_retrieval = BM25Retrieval(save_path=bm25_path)
+    ready_pickle_db(pickle_path)
+    yield bm25_retrieval
+    # teardown
+    if os.path.exists(bm25_path):
+        os.remove(bm25_path)
+    if os.path.exists(pickle_path):
+        os.remove(pickle_path)
+
+
+def test_duplicate_check(just_bm25_retrieval):
+    assert just_bm25_retrieval.duplicate_check(TEST_DB_ORIGIN) == TEST_DB_ORIGIN_RESULT
+
+
+def test_is_created(just_bm25_retrieval):
+    """
+    For 'is_created'
+    1. db_instance_list가 비어있을 때, list에 db instance가 잘 들어가는가? 해당 db를 잘 return하는가?
+    2. db_instance_list가 비어있지 않고, db_origin이 이미 존재하는 경우, 해당 db를 잘 return하는가?
+    3. db_instance_list가 비어있지 않고, db_origin이 존재하지 않는 경우, list에 db instance가 잘 들어가는가? 해당 db를 잘 return하는가?
+    """
+    # 0. reset db_instance_list for test
+    just_bm25_retrieval.db_instance_list = []
+    # 1. If 'db_instance_list' is empty
+    first_instance = just_bm25_retrieval.is_created(db_type=TEST_DB_ORIGIN[0]['db_type'],
+                                                    db_path=TEST_DB_ORIGIN[0]['db_path'])
+    assert first_instance == just_bm25_retrieval.db_instance_list[0]
+    # 2. db_instance_list is not empty, db_origin already exists
+    second_instance = just_bm25_retrieval.is_created(db_type=TEST_DB_ORIGIN[0]['db_type'],
+                                                     db_path=TEST_DB_ORIGIN[0]['db_path'])
+    assert second_instance == just_bm25_retrieval.db_instance_list[0]
```

Comment:
Then first_instance and second_instance is equal, right? Why don't you assert first_instance == second_instance, too?